centreon/src/Core/AgentConfiguration/Application/UseCase/FindAgentConfiguration/FindAgentConfiguration.php

```diff
@@ -0,0 +1,124 @@
+     * @param ReadMonitoringServerRepositoryInterface $readMonitoringServerRepository repository to read monitoring servers
+     * @param ReadAccessGroupRepositoryInterface $readAccessGroupRepository repository to read access groups
+     */
+    public function __construct(
+        private readonly ContactInterface $user,
+        private readonly ReadAgentConfigurationRepositoryInterface $readRepository,
+        private readonly ReadMonitoringServerRepositoryInterface $readMonitoringServerRepository,
+        private readonly ReadAccessGroupRepositoryInterface $readAccessGroupRepository,
+    ) {
+    }
+
+    /**
+     * Retrieves an agent configuration with associated pollers.
+     *
+     * @param int $agentConfigurationId
+     *
+     * @return FindAgentConfigurationResponse|ResponseStatusInterface
+     */
+    public function __invoke(int $agentConfigurationId): FindAgentConfigurationResponse|ResponseStatusInterface {
+        $this->info(
+            'Find agent configuration',
+            [
+                'user_id' => $this->user->getId(),
+                'agent_configuration_id' => $agentConfigurationId,
+            ]
+        );
+
+        try {
+            if (null === $agentConfiguration = $this->readRepository->find($agentConfigurationId)) {
+                $this->error(
+                    'Agent configuration not found',
+                    ['agent_configuration_id' => $agentConfigurationId]
+                );
+
+                return new NotFoundResponse('Agent Configuration');
+            }
+
+            $this->info(
+                'Retrieved agent configuration',
+                ['agent_configuration_id' => $agentConfigurationId]
+            );
+
+            $pollers = $this->readRepository->findPollersByAcId($agentConfigurationId);
+
+            /**
+             * Check if the non-admin user has access to all the pollers in the agent configuration.
+             * If not, return a NotFoundResponse.
+             */
+            if (! $this->user->isAdmin()) {
+                $pollerIds = array_map(
+                    static fn(Poller $poller): int => $poller->id,
+                    $pollers
+                );
+                $validPollerIds = $this->readMonitoringServerRepository->existByAccessGroups(
+                    $pollerIds,
+                    $this->readAccessGroupRepository->findByContact($this->user)
+                );
+
+                if ([] !== array_diff($pollerIds, $validPollerIds)) {
+                    $this->debug(
+                        'User does not have the correct access groups for pollers',
+                        [
+                            'user_id' => $this->user->getId(),
+                            'poller_ids' => array_diff($pollerIds, $validPollerIds),
+                        ]
+                    );
+
+                    return new NotFoundResponse('Agent Configuration');
+                }
```

Comment:
This part is also related to user permissions.

It could be relevant to use another voter attribute to check if current user has correct access groups for pollers and return a 404 code if he hasn't.

Your use case would no longer need to be user-aware.

Reply:
Also, the response message is confusing and does not bring any information.

Something like: 

```
Agent Configuration with id #xxx could not be found
```

would be a minimum IMO

Reply:
TBD in backend discussions point